### PR TITLE
If the pattern is a simple String, do a String compare and not ReguarExpresion match

### PR DESCRIPTION
**Pull request by shmuelk**
## Description
A check is made on each route pattern to see if it contains a capture group, (:key or (\w) ) or any regular expression matching (*+[]\.) characters. If it doesn't, at run time a String compare for non-partial match Route elements or a String.hasPrefix is done, in lieu of a RegularExpression
match.

## Motivation and Context
Improve performance. On theTechEmpower hello world benchmark perfrmance improves on Ubuntu 14.04 by about 18% and Ubuntu 16.04 by about 27%.

See issue IBM-Swift/Kitura#940

## How Has This Been Tested?
@djones6 Tested to get the performance numbers.
I ran the Kitura unit tests on Linux and macOS

## Checklist:
- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/Kitura/RouteRegex.swift [hunk 1]
```diff
@@ -52,10 +55,17 @@ public class RouteRegex {
...
+    ///            this is a simple String compare, and array of keys
+    internal func buildRegex(fromPattern: String?, allowPartialMatch: Bool = false) -> (RegularExpressionType?, Bool, [String]?) {
         guard let pattern = fromPattern else {
-            return (nil, nil)
+            return (nil, false, nil)
+        }
+        
+        // Check and see if the pattern is a simple string (no captures and not a regular expression)
+        let range = NSMakeRange(0, pattern.characters.count)
+        guard let _ = simpleStringChecker.firstMatch(in: pattern, options: [], range: range) else {
```
Comment (vadimeisenbergibm) [inline Sources/Kitura/RouteRegex.swift]: The following can be achieved by
```swift
let complexRouteCharacters = CharacterSet(charactersIn: "*.:+?()[]\")

if pattern.rangeOfCharacter(from: complexRouteCharacters) != nil {
  return (nil, true, nil)
} 
```
Maybe we should measure which method is faster.

File: Sources/Kitura/RouterElement.swift [hunk 2]
```diff
@@ -124,6 +135,44 @@ class RouterElement {
...
+    /// - Parameter response: the router response
+    /// - Parameter next: the closure for the next execution block
+    private func performSimpleMatch(path: String, request: RouterRequest, response: RouterResponse, next: @escaping () -> Void) {
+        let pathToMatch = path.isEmpty ? "/" : path
+        var matched: Bool
+        let matchedPath: String
+        
+        if allowPartialMatch {
+            matched = pathToMatch.hasPrefix(pattern!)
+            if matched && pattern! != "/" {
```
Comment (vadimeisenbergibm) [inline Sources/Kitura/RouterElement.swift]: I would add a `guard let pattern = pattern else` in the beginning of the function and would not use `pattern!`

File: Sources/Kitura/RouteRegex.swift (hunk 1 above)
Comment (shmuelk) [inline Sources/Kitura/RouteRegex.swift]: Fixed as we discussed.

File: Sources/Kitura/RouterElement.swift (hunk 2 above)
Comment (shmuelk) [inline Sources/Kitura/RouterElement.swift]: Done.

**Pull request merged**